src/ui/wxWidgets/YubiMixin.cpp

```diff
@@ -71,11 +73,11 @@ bool YubiMixin::IsYubiInserted() const
 void YubiMixin::HandlePollingTimer()
 {
   // Show Yubi controls when inserted first time:
-  if (yubiExists()) {
+  if (yubiExists() && !m_btn->IsShown() && !m_status->IsShown()) {
     wxWindow *parent = nullptr; // assume both have same parent
-    if (m_btn != nullptr) {m_btn->Show(true); parent = m_btn->GetParent();}
-    if (m_status != nullptr) {m_status->Show(true); parent = m_btn->GetParent();}
-    if (parent != nullptr) parent->Layout();
+    m_btn->Show(true); parent = m_btn->GetParent();
+    m_status->Show(true); parent = m_btn->GetParent();
```

Comment:
Unless `Show` modifies the value of `m_btn->GetParent` the second `parent = m_btn->GetParent()` is now pointless.